tools/SwiftSyntax/RawSyntax.swift

```diff
@@ -43,53 +53,57 @@ struct SyntaxNodeId: Hashable, Codable {
   }
 }
 
+/// The data that is specific to a tree or token node
+fileprivate indirect enum RawSyntaxData {
```

Comment:
Why this is declared as `indirect`? I know the original `RawSyntax` was also `indirect` @harlanhaskins, what was the reason? maybe `size(of: RawSyntax)`?

Reply:
At the time of writing, it ensured both cases were heap allocated and that a copy = a retain, so we could share raw data. Pretty sure that is still the case.

Reply:
It's because `.node` has `RawSyntax` as its member which again has `RawSyntaxData` as one of its members. That's a cycle and thus the object size wouldn't be determined at runtime if the enum is not `indirect`. 

Or TL;DR: It doesn't compile if the enum isn't `indirect`.

Reply:
It should still compile. It’s already got a layer of indirection via the array of `RawSyntax?`.

Reply:
@ahoppen It does compile because `[RawSyntaxData]` provides indirect semantics.

Reply:
> At the time of writing, it ensured both cases were heap allocated and that a copy = a retain, so we could share raw data.
Makes sense. Thanks :)

Reply:
> It does compile because [RawSyntaxData] provides indirect semantics.

Hmm. I was pretty sure it didn't compile at some point without `indirect`. But maybe that was just my imagination. @harlanhaskins's explanation makes sense. I'll leave it as is.